recirq/engine_utils.py

```diff
@@ -226,7 +217,7 @@ async def run_async(self, program: 'cirq.Circuit',
         return results[0]
 
 
-@dataclass(frozen=True)
+@dataclass
```

Comment:
Ah I see. We were cheating before because dataclass(frozen=True) can't enforce immutability if you sneak a dictionary into it. If we cared about this class, I'd probably migrate it differently; but new code should use `cg.ProcessorRecord` subclasses if you need a frozen dataclass serializable record of what processor you're using